### PR TITLE
Refactor BatchedANIDataset and create API for splittting datasets

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
This pull request **introduces 2 alerts** when merging d1ecd93d75b10e8ac552f506895bdfba7082e2cf into c630982fd0fb39bbb732b112316e49fab74dcbba - [view on LGTM.com](https://lgtm.com/projects/g/aiqm/torchani/rev/pr-ca50eda0d77df4a8d746f0f9c211678d8d14e9fd)

**new alerts:**

* 2 for Unused local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by zasdfgbnm** [comment]
This pull request **introduces 4 alerts** when merging 912a8855ab13862d5eeb02d639f5a4430b3e2832 into c630982fd0fb39bbb732b112316e49fab74dcbba - [view on LGTM.com](https://lgtm.com/projects/g/aiqm/torchani/rev/pr-2e240b128742c361fc9e147e208ae7ea400dcbf4)

**new alerts:**

* 3 for Non\-callable called
* 1 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by zasdfgbnm** [comment]
Also be aware that I am deprecating `torchani.data.BatchedANIDataset` in favor of `torchani.data.load_ani_dataset`, the usage are the same, except that `load_ani_dataset` takes an additional optional argument `split` which guide load_ani_dataset to split the whole dataset.

File: examples/nnp_training_ignite.py [hunk 1]
```diff
@@ -102,13 +101,9 @@ def atomic():
...
 ###############################################################################
 # Now load training and validation datasets into memory.
-training = torchani.data.BatchedANIDataset(
-    training_path, consts.species_to_tensor, batch_size, device=device,
-    transform=[energy_shifter.subtract_from_dataset])
-
-validation = torchani.data.BatchedANIDataset(
-    validation_path, consts.species_to_tensor, batch_size, device=device,
-    transform=[energy_shifter.subtract_from_dataset])
+validation, training = torchani.data.load_ani_dataset(
```
Comment (farhadrgh) [inline examples/nnp_training_ignite.py]: Why not make the order consistent "training, validation" and then keep split=[0.8, None] ?

Comment (zasdfgbnm) [reply examples/nnp_training_ignite.py]: fixed

**Pull request merged**